hermes/probe/mock/mock_gcs.go

```diff
@@ -0,0 +1,88 @@
+package mock
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+)
+
+type fakeBucket struct {
+	attrs   *storage.BucketAttrs
+	objects map[string][]byte
+}
+
+type fakeClient struct {
+	stiface.Client
+	buckets map[string]*fakeBucket
+}
+
+type fakeBucketHandle struct {
+	stiface.BucketHandle
+	c    *fakeClient
+	name string
+}
+
+type fakeObjectHandle struct {
+	stiface.ObjectHandle
+	c          *fakeClient
+	bucketName string
+	name       string
+}
+
+type fakeWriter struct {
+	stiface.Writer
+	obj fakeObjectHandle
+	buf bytes.Buffer
+}
+
```

Comment:
Add object iterator for list objects.
```suggestion

type fakeObjectIterator struct {                                                                                                                                    
	idx  int                                                                                                                                                 
	objects map[int]*storage.ObjectAttrs                                                                                                                        
	stiface.ObjectIterator                                                                                                                                      
}  

```